ld fails in coffcode.h line 842 in handle_COMDAT

http://sourceware.org/bugzilla/show_bug.cgi?id=6962

===================================================================
RCS file: /cvs/src/src/bfd/ChangeLog,v
retrieving revision 1.5169
retrieving revision 1.5170
diff -u -r1.5169 -r1.5170
--- src/bfd/ChangeLog	2010/10/29 12:10:19	1.5169
+++ src/bfd/ChangeLog	2010/10/29 13:20:03	1.5170
@@ -1,3 +1,9 @@
+2010-10-29  Nick Clifton  <nickc@redhat.com>
+
+	PR bintils/6962
+	* coffcode.h (handle_COMDAT): Only check the base type when
+	looking for a section symbol.
+
 2010-02-24  Alan Modra  <amodra@gmail.com>
 
 	PR binutils/10858
===================================================================
RCS file: /cvs/src/src/bfd/coffcode.h,v
retrieving revision 1.169
retrieving revision 1.170
diff -u -r1.169 -r1.170
--- src/bfd/coffcode.h	2010/09/22 14:19:35	1.169
+++ src/bfd/coffcode.h	2010/10/29 13:20:04	1.170
@@ -956,7 +956,7 @@
 
 		if (! ((isym.n_sclass == C_STAT
 			|| isym.n_sclass == C_EXT)
-		       && isym.n_type == T_NULL
+		       && BTYPE (isym.n_type) == T_NULL
 		       && isym.n_value == 0))
 		  abort ();
 
-- 
